tests/test_integration_http.py

```diff
@@ -625,7 +625,12 @@ def test_capture_minidump(cmake, httpserver):
     ],
 )
 @pytest.mark.parametrize("proxy_status", [(["off"]), (["on"])])
-def test_capture_proxy(cmake, httpserver, run_args, proxy_status):
+@pytest.mark.parametrize("proxy_auth", [(["off"]), (["on"])])
+@pytest.mark.parametrize("proxy_IPv", [(["4"]), (["6"])])
+@pytest.mark.parametrize("proxy_from_env", [(["proxy-from-env"]), ([""])])
```

Comment:
This test is becoming exponentially larger. It might make sense to split apart some test cases to not cover the full matrix at once. There are still some untested cases:

- [ ] What if `proxy-from-env` but the env. variable is empty?  -> expected is that if there was a manually set proxy, it gets used.
- [ ] Test whether`proxy-from-env` takes precedence over manually set proxy if both are given.